torch/functional.py

```diff
@@ -330,9 +330,38 @@ def meshgrid(*tensors: Union[Tensor, List[Tensor]]) -> Tuple[Tensor, ...]:
         return _meshgrid(*tensors)
 else:
     def meshgrid(*tensors):
-        r"""Take :math:`N` tensors, each of which can be either scalar or 1-dimensional
-        vector, and create :math:`N` N-dimensional grids, where the :math:`i` :sup:`th` grid is defined by
-        expanding the :math:`i` :sup:`th` input over dimensions defined by other inputs.
+        r"""Creates grids of coordinates specified by the 1D input tensors.
+
+        This is helpful when you want to visualize data over some
+        range of inputs. See below for plotting example.
+
+        Given :math:`N` 1D tensors :math:`T_0 \ldots T_{N-1}` as
+        inputs with corresponding sizes :math:`S_0 \ldots S_{N-1}`,
+        this creates :math:`N` N-dimensional tensors :math:`G_0 \ldots
+        G_{N-1}`, each with shape :math:`(S_0, ..., S_{N-1})` where
+        the output :math:`G_i` is constructed by expanding :math:`T_i`
+        to the result shape.
+
+        .. note::
+            0D inputs are treated equivalently to 1D inputs of a
+            single element.
+
+        .. warning::
+
+            This behaves differently from `numpy.meshgrid`.
+            `numpy.meshgrid` supports an indexing argument, defaulting
```

Comment:
This can go in now if this warning is updated. I think it can be simplified to something like:

"torch.meshgrid has the same behavior as calling numpy.meshgrid(..., indexing='ij'), and in the future torch.meshgrid will also support the indexing argument."

Reply:
Link to the actual issue in the following paragraph seems cool

Reply:
works for me!